### PR TITLE
perf: speed up percent_decode using bulk run-copying

### PR DESCRIPTION
**Pull request by jbergstroem**
Speed up `ada::unicode::percent_decode`, the routine that decodes `%XX` escapes when parsing hosts (via `to_ascii`) and search-param values.

The decoded output is never longer than the input, size the result buffer once and copy plain runs in bulk; the same approach `form_urlencoded_decode` already uses.

## What changed

- Allocate the output once at `input.size()` and shrink to the final length at the end, instead of `reserve` +  per-byte `+=`.
- Plain (non-`%`) spans are located with `std::memchr('%')` and copied with a single `std::memcpy`, rather than appending one character at a time.
- `%XX` runs are decoded through the existing `unhex_table` (two loads + an OR to validate), which is above the function so both `percent_decode` and `form_urlencoded_decode` share it.
- Add `benchmarks/percent_decode.cpp`, mirroring `percent_encode.cpp`.
- Increase coverage in the test suite to pick up introduced branches.

## Benchmark results

Measured with a Apple M5 (`benchmarks/percent_decode`, `--benchmark_min_time=1s`):

| Benchmark | Before | After | Improvement |
|-----------|--------|-------|-------------|
| `Decode` time/url | ~54.3 ns | ~36.4 ns | **1.49×** |
| `Decode` throughput | ~1.03 GB/s | ~1.54 GB/s | **1.49×** |

## Testing

- Full test suite passes
- `clang-format`, `clang-tidy` introduces no new findings
- Verified output to be byte-identical to the previous implementation over fuzzed inputs plus a few hand-picked edge cases (`%`, `%2`, `%%41`,`%g0`, trailing `%`, doubly-encoded, some utf-8 strings)

Edit: side-note, just noticed https://github.com/ada-url/ada/pull/1124 which might be a longer term solution; feel free to close this if thats the preferred path.

### CODE REVIEW

**Comment by anonrig** [comment]
Nice work!

File: src/unicode.cpp [hunk 1]
```diff
@@ -504,6 +471,60 @@ constexpr static std::array<uint8_t, 256> unhex_table = []() consteval {
   return t;
 }();
 
+std::string percent_decode(const std::string_view input, size_t first_percent) {
```
Comment (anonrig) [inline src/unicode.cpp:458]: Can you put the method to the previous place so the git diff is nicer?

**Comment by anonrig** [comment]
>Edit: side-note, just noticed https://github.com/ada-url/ada/pull/1124 which might be a longer term solution; feel free to close this if thats the preferred path.

It has regressions... That's why we couldn't land it.

**Review by anonrig — APPROVED** [review]
